Upgrade Mysql test container from 8 to 9

### Description

there is this kind of error when upgrading to 9.5:

```
Build step io.quarkus.datasource.deployment.devservices.DevServicesDatasourceProcessor#launchDatabases threw an exception: java.lang.RuntimeException: org.testcontainers.containers.ContainerLaunchException: Container startup failed for image mirror.gcr.io/mysql:9.5
	at io.quarkus.datasource.deployment.devservices.DevServicesDatasourceProcessor.startDevDb(DevServicesDatasourceProcessor.java:401)
	at io.quarkus.datasource.deployment.devservices.DevServicesDatasourceProcessor.launchDatabases(DevServicesDatasourceProcessor.java:142)
	at java.base/java.lang.invoke.MethodHandle.invokeWithArguments(MethodHandle.java:732)
	at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:872)
	at io.quarkus.builder.BuildContext.run(BuildContext.java:255)
	at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
	at org.jboss.threads.EnhancedQueueExecutor$Task.doRunWith(EnhancedQueueExecutor.java:2651)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2630)
	at org.jboss.threads.EnhancedQueueExecutor.runThreadBody(EnhancedQueueExecutor.java:1622)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1589)
	at java.base/java.lang.Thread.run(Thread.java:840)
	at org.jboss.threads.JBossThread.run(JBossThread.java:501)
Caused by: org.testcontainers.containers.ContainerLaunchException: Container startup failed for image mirror.gcr.io/mysql:9.5
	at org.testcontainers.containers.GenericContainer.doStart(GenericContainer.java:351)
	at org.testcontainers.containers.GenericContainer.start(GenericContainer.java:322)
	at io.quarkus.devservices.mysql.deployment.MySQLDevServicesProcessor$1.lambda$startDatabase$0(MySQLDevServicesProcessor.java:91)
	at java.base/java.util.Optional.orElseGet(Optional.java:364)
	at io.quarkus.devservices.mysql.deployment.MySQLDevServicesProcessor$1.startDatabase(MySQLDevServicesProcessor.java:107)
	at io.quarkus.datasource.deployment.devservices.DevServicesDatasourceProcessor.startDevDb(DevServicesDatasourceProcessor.java:327)
	... 11 more
Caused by: org.rnorth.ducttape.RetryCountExceededException: Retry limit hit with exception
	at org.rnorth.ducttape.unreliables.Unreliables.retryUntilSuccess(Unreliables.java:88)
	at org.testcontainers.containers.GenericContainer.doStart(GenericContainer.java:336)
	... 16 more
Caused by: org.testcontainers.containers.ContainerLaunchException: Could not create/start container
	at org.testcontainers.containers.GenericContainer.tryStart(GenericContainer.java:556)
	at org.testcontainers.containers.GenericContainer.lambda$doStart$0(GenericContainer.java:346)
	at org.rnorth.ducttape.unreliables.Unreliables.retryUntilSuccess(Unreliables.java:81)
	... 17 more
Caused by: java.lang.IllegalStateException: Wait strategy failed. Container exited with code 1
	at org.testcontainers.containers.GenericContainer.tryStart(GenericContainer.java:525)
	... 19 more
Caused by: java.lang.IllegalStateException: Container is started, but cannot be accessed by (JDBC URL: jdbc:mysql://localhost:32776/mysql), please check container logs
	at org.testcontainers.containers.JdbcDatabaseContainer.waitUntilContainerStarted(JdbcDatabaseContainer.java:210)
	at org.testcontainers.containers.GenericContainer.tryStart(GenericContainer.java:492)
	... 19 more
Caused by: java.sql.SQLException: Could not create new connection
	at org.testcontainers.containers.JdbcDatabaseContainer.createConnection(JdbcDatabaseContainer.java:296)
	at org.testcontainers.containers.JdbcDatabaseContainer.createConnection(JdbcDatabaseContainer.java:252)
	at org.testcontainers.containers.JdbcDatabaseContainer.waitUntilContainerStarted(JdbcDatabaseContainer.java:192)
	... 20 more
```

the password requirement is not new in fact:

```
podman run mirror.gcr.io/mysql:8.4
Trying to pull mirror.gcr.io/mysql:8.4...
Getting image source signatures
Copying blob 23eb2baa39f3 done   | 
Copying blob ad9d782f3f87 skipped: already exists  
Copying blob 88358ea2a37f skipped: already exists  
Copying blob 3709f9999ba9 skipped: already exists  
Copying blob 98f63f165ac1 skipped: already exists  
Copying blob 100b56c3fd28 skipped: already exists  
Copying blob 08d96bdd8a50 done   | 
Copying blob bec4df3fa85f done   | 
Copying blob c68ab04cc1e9 done   | 
Copying blob 8c32caf90444 done   | 
Copying config 54c6e074ef done   | 
Writing manifest to image destination
2026-01-15 10:02:54+00:00 [Note] [Entrypoint]: Entrypoint script for MySQL Server 8.4.7-1.el9 started.
2026-01-15 10:02:54+00:00 [Note] [Entrypoint]: Switching to dedicated user 'mysql'
2026-01-15 10:02:54+00:00 [Note] [Entrypoint]: Entrypoint script for MySQL Server 8.4.7-1.el9 started.
2026-01-15 10:02:54+00:00 [ERROR] [Entrypoint]: Database is uninitialized and password option is not specified
    You need to specify one of the following as an environment variable:
    - MYSQL_ROOT_PASSWORD
    - MYSQL_ALLOW_EMPTY_PASSWORD
    - MYSQL_RANDOM_ROOT_PASSWORD
```

so it was already configured.

--- a/pom.xml
+++ b/pom.xml
@@ -255,7 +255,7 @@
         <localstack.container.image>mirror.gcr.io/localstack/localstack:4.9.2</localstack.container.image>
         <lra-coordinator.container.image>quay.io/jbosstm/lra-coordinator:7.0.1.Final-3.8.3</lra-coordinator.container.image>
         <minio.container.image>mirror.gcr.io/minio/minio:RELEASE.2025-07-23T15-54-02Z-cpuv1</minio.container.image>
-        <mysql.container.image>mirror.gcr.io/mysql:8.4</mysql.container.image>
+        <mysql.container.image>mirror.gcr.io/mysql:9.5</mysql.container.image>
         <mariadb.container.image>mirror.gcr.io/mariadb:12.1</mariadb.container.image>
         <mongodb.container.image>mirror.gcr.io/mongo:7.0</mongodb.container.image>
         <nats.container.image>mirror.gcr.io/nats:2.11.6</nats.container.image>

